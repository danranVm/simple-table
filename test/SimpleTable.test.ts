import { shallowMount,mount } from "@vue/test-utils";
import { expect, test } from "vitest";
import SimpleTable from "../src/components/SimpleTable";
import { nextTick } from 'vue';
import * as _ from 'lodash';
import { DATA, COLUMN } from "../src/static/data"

test("mount component", () => {
  const wrapper:any = shallowMount(SimpleTable, {
    props: {
      tableData : DATA,
      columns : COLUMN
    },
  });
  expect(wrapper.html()).toMatchSnapshot();
  
});

// 数据为空
test('empty body', async () => {
  const wrapper:any = shallowMount(SimpleTable, {
    props: {
      tableData : DATA,
      columns : COLUMN
    }
  });
  expect(wrapper.find('.empty-table').exists() === false).toBe(true)
  await wrapper.setProps({
    tableData: [],
  })
  expect(wrapper.find('.empty-table').exists() === true).toBe(true)
})

// 点击排序
test('test click sort', async () => {
  const sortData = DATA.slice(0, 2);
  const copyData = _.cloneDeep(sortData);

  const wrapper = shallowMount(SimpleTable, {
    props: {
      tableData : copyData,
      columns : COLUMN
    },
  })
  const sortColumn = wrapper.find('.table-header-sort')
  const body:any = wrapper.find('.table-body')
  sortColumn.trigger('click')
  await nextTick()

  expect(body.html().includes('<tr class="tr0" mark="row0">')).toBe(true)

  sortColumn.trigger('click')
  await nextTick()

  const body1:any = wrapper.find('.table-body')
  expect(body1.html().includes('<tr class="tr0" mark="row1">')).toBe(true)
})
