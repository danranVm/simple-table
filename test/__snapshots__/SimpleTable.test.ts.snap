// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div>
  <table class=\\"table\\">
    <thead class=\\"table-header\\">
      <th>ID<span class=\\"table-header-sort\\"><i class=\\"iconfont icon-caret-up table-header-sort-up\\"></i><i class=\\"iconfont icon-caret-down table-header-sort-down\\"></i></span></th>
      <th>名称<span class=\\"table-header-sort\\"><i class=\\"iconfont icon-caret-up table-header-sort-up\\"></i><i class=\\"iconfont icon-caret-down table-header-sort-down\\"></i></span></th>
      <th>性别<span class=\\"table-header-sort\\"><i class=\\"iconfont icon-caret-up table-header-sort-up\\"></i><i class=\\"iconfont icon-caret-down table-header-sort-down\\"></i></span></th>
      <th>年龄<span class=\\"table-header-sort\\"><i class=\\"iconfont icon-caret-up table-header-sort-up\\"></i><i class=\\"iconfont icon-caret-down table-header-sort-down\\"></i></span></th>
    </thead>
    <tbody class=\\"table-body\\">
      <tr class=\\"tr0\\" mark=\\"row0\\">
        <td>0</td>
        <td>a</td>
        <td>male</td>
        <td>12</td>
      </tr>
      <tr class=\\"tr1\\" mark=\\"row1\\">
        <td>1</td>
        <td>b</td>
        <td>female</td>
        <td>64</td>
      </tr>
      <tr class=\\"tr2\\" mark=\\"row2\\">
        <td>2</td>
        <td>c</td>
        <td>male</td>
        <td>5</td>
      </tr>
      <tr class=\\"tr3\\" mark=\\"row3\\">
        <td>3</td>
        <td>d</td>
        <td>female</td>
        <td>21</td>
      </tr>
      <tr class=\\"tr4\\" mark=\\"row4\\">
        <td>4</td>
        <td>e</td>
        <td>female</td>
        <td>75</td>
      </tr>
      <tr class=\\"tr5\\" mark=\\"row5\\">
        <td>5</td>
        <td>f</td>
        <td>male</td>
        <td>25</td>
      </tr>
      <tr class=\\"tr6\\" mark=\\"row6\\">
        <td>6</td>
        <td>g</td>
        <td>male</td>
        <td>42</td>
      </tr>
      <tr class=\\"tr7\\" mark=\\"row8\\">
        <td>8</td>
        <td>h</td>
        <td>female</td>
        <td>13</td>
      </tr>
      <tr class=\\"tr8\\" mark=\\"row9\\">
        <td>9</td>
        <td>wv</td>
        <td>female</td>
        <td>56</td>
      </tr>
      <tr class=\\"tr9\\" mark=\\"row10\\">
        <td>10</td>
        <td>ki</td>
        <td>male</td>
        <td>76</td>
      </tr>
    </tbody>
  </table>
  <div class=\\"table-pagination\\">
    <ul>
      <li class=\\"table-pagination_prev\\">上一页</li>
      <li class=\\"table-pagination_next\\">下一页</li>
      <li class=\\"table-pagiantion_info\\">共 33 条</li>
      <li class=\\"table-pagiantion_info\\">第 1 页</li>
      <li class=\\"table-pagiantion_info\\">共 4 页</li>
      <li class=\\"table-pagiantion_info\\"><input class=\\"inputPage\\" type=\\"number\\"><span>跳转</span></li>
    </ul>
  </div>
</div>"
`;
